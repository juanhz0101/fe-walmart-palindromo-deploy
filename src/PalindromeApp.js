import React, { useState } from 'react'
import ProductGrid from './components/products/ProductGrid'
import AddSearch from './components/products/AddSearch'

function PalindromeApp({ defaultQuery = { query: '', pagination:'/products?page=1&limit=20'} }) {

    const [query, setQuery] = useState(defaultQuery)
    
    return (
        <>
            <h2>Lider Palindromo</h2>
            <hr />
            <AddSearch setQuery={setQuery} /> 
            <ProductGrid query={query} setQuery={setQuery} /> 
            
        </>
    )
}

export default PalindromeApp