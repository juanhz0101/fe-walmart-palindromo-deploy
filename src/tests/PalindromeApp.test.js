import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import PalindromeApp from "../PalindromeApp";

describe('Pruebas en <PalindromeApp />', () => {

    test('debe de mostrar <PalindromeApp /> correctamente', () => {
        
        const wrapper = shallow( <PalindromeApp />);
        expect( wrapper ).toMatchSnapshot();
    })

    test('debe mostrar una lista de productos', () => {
        
        const defaultQuery = { query: '', pagination:'/products?page=1&limit=20'};
        const wrapper = shallow( <PalindromeApp defaultQuery={defaultQuery} />)
        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('ProductGrid').length ).toBe( 1 )

    })

})