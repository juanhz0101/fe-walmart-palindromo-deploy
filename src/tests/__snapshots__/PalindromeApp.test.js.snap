// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <PalindromeApp /> debe de mostrar <PalindromeApp /> correctamente 1`] = `
<Fragment>
  <h2>
    Lider Palindromo
  </h2>
  <hr />
  <AddSearch
    setQuery={[Function]}
  />
  <ProductGrid
    query={
      Object {
        "pagination": "/products?page=1&limit=20",
        "query": "",
      }
    }
    setQuery={[Function]}
  />
</Fragment>
`;

exports[`Pruebas en <PalindromeApp /> debe mostrar una lista de productos 1`] = `
<Fragment>
  <h2>
    Lider Palindromo
  </h2>
  <hr />
  <AddSearch
    setQuery={[Function]}
  />
  <ProductGrid
    query={
      Object {
        "pagination": "/products?page=1&limit=20",
        "query": "",
      }
    }
    setQuery={[Function]}
  />
</Fragment>
`;
